builder/beacon_client.go

```diff
@@ -256,3 +286,59 @@ func fetchBeacon(url string, dst any) error {
 	log.Info("fetched", "url", url, "res", dst)
 	return nil
 }
+
+// SubscribeToPayloadAttributesEvents subscribes to payload attributes events to validate fields such as prevrandao and withdrawals
+func (b *BeaconClient) SubscribeToPayloadAttributesEvents(payloadAttrC chan types.BuilderPayloadAttributes) {
+	payloadAttributesResp := &struct {
+		Version string `json:"version"`
+		Data    struct {
+			ProposalSlot      uint64      `json:"proposal_slot,string"`
+			ParentBlockHash   common.Hash `json:"parent_block_hash"`
+			PayloadAttributes struct {
+				Timestamp             uint64                `json:"timestamp,string"`
+				PrevRandao            common.Hash           `json:"prev_randao"`
+				SuggestedFeeRecipient common.Address        `json:"suggested_fee_recipient"`
+				Withdrawals           []*capella.Withdrawal `json:"withdrawals"`
+			} `json:"payload_attributes"`
+		} `json:"data"`
+	}{}
+
+	eventsURL := fmt.Sprintf("%s/eth/v1/events?topics=payload_attributes", b.endpoint)
+	log.Info("subscribing to payload_attributes events")
+
+	for {
+		client := sse.NewClient(eventsURL)
+		err := client.SubscribeRaw(func(msg *sse.Event) {
+			err := json.Unmarshal(msg.Data, &payloadAttributesResp)
```

Comment:
Is this a double reference - && struct?
I'd much prefer if we had the structure defined as a standalone for ease of reasoning